File: passwordcomplexity.cpp
// #include <iostream>
// #include <string>
// #include <regex>
#include<bits/stdc++.h>

using namespace std;

bool containsUppercase(const string& password) {
    return regex_search(password, regex("[A-Z]"));
}

bool containsLowercase(const string& password) {
    return regex_search(password, regex("[a-z]"));
}

bool containsDigit(const string& password) {
    return regex_search(password, regex("[0-9]"));
}

bool containsSpecialChar(const string& password) {
    return regex_search(password, regex("[!@#$%^&*(),.?\":{}|<>]"));
}

string assessPasswordStrength(const string& password) {
    int strengthScore = 0;
    string feedback = "";

    if (password.length() >= 8) {
        strengthScore++;
    } else {
        feedback += "Password should be at least 8 characters long.\n";
    }

    if (containsUppercase(password)) {
        strengthScore++;
    } else {
        feedback += "Password should include at least one uppercase letter.\n";
    }

    if (containsLowercase(password)) {
        strengthScore++;
    } else {
        feedback += "Password should include at least one lowercase letter.\n";
    }

    if (containsDigit(password)) {
        strengthScore++;
    } else {
        feedback += "Password should include at least one number.\n";
    }

    if (containsSpecialChar(password)) {
        strengthScore++;
    } else {
        feedback += "Password should include at least one special character.\n";
    }

    if (strengthScore == 5) {
        feedback += "Your password is strong.\n";
    } else if (strengthScore >= 3) {
        feedback += "Your password is moderate.\n";
    } else {
        feedback += "Your password is weak.\n";
    }

    return feedback;
}

int main() {
    string password;
    cout << "Enter your password: ";
    cin >> password;

    string feedback = assessPasswordStrength(password);

    cout << "\nPassword Feedback:\n" << feedback;

    return 0;
}

File: passwordcomplexity.py
import re

def assess_password_strength(password):
    strength_criteria = {
        "length": len(password) >= 8,
        "uppercase": re.search(r"[A-Z]", password) is not None,
        "lowercase": re.search(r"[a-z]", password) is not None,
        "numbers": re.search(r"[0-9]", password) is not None,
        "special_characters": re.search(r"[!@#$%^&*(),.?\":{}|<>]", password) is not None
    }

    strength_score = sum(strength_criteria.values())
    
    feedback = []
    if not strength_criteria["length"]:
        feedback.append("Password should be at least 8 characters long.")
    if not strength_criteria["uppercase"]:
        feedback.append("Password should include at least one uppercase letter.")
    if not strength_criteria["lowercase"]:
        feedback.append("Password should include at least one lowercase letter.")
    if not strength_criteria["numbers"]:
        feedback.append("Password should include at least one number.")
    if not strength_criteria["special_characters"]:
        feedback.append("Password should include at least one special character.")
    
    if strength_score == 5:
        feedback.append("Your password is strong.")
    elif 3 <= strength_score < 5:
        feedback.append("Your password is moderate.")
    else:
        feedback.append("Your password is weak.")
    
    return feedback

def main():
    password = input("Enter your password: ")
    feedback = assess_password_strength(password)
    
    print("\nPassword Feedback:")
    for comment in feedback:
        print(f"- {comment}")

if __name__ == "__main__":
    main()
